libcodechecker/analyze/store_handler.py

```diff
@@ -0,0 +1,526 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+
+from hashlib import sha256
+import json
+import os
+import tempfile
+import zlib
+
+import sqlalchemy
+
+import shared
+from codeCheckerDBAccess.ttypes import *
+
+from libcodechecker.analyze import skiplist_handler
+from libcodechecker.logger import LoggerFactory
+from libcodechecker.orm_model import *
+
+LOG = LoggerFactory.get_new_logger('STORE HANDLER')
+
+
+def process_compile_db(compile_db):
+    """
+    Create a dict where the source file name (absolute path) is the key and
+    the correspoding compilation command is the value.
+    """
+    compile_db = os.path.abspath(compile_db)
+    comp_cmds = {}
+
+    if not os.path.exists(compile_db):
+        LOG.warning("Compilation command file '" + compile_db +
+                    "' is missing.")
+        LOG.warning("Update mode will not work, without compilation "
```

Comment:
Somewhere we should have a ticket to extend the plist with a working directory entry to overcome this issue in the long term.